src/components/autocomplete-grid/autocomplete-grid-button/autocomplete-grid-button.ts

```diff
@@ -0,0 +1,115 @@
+import type { CSSResultGroup, PropertyValues } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+
+import { SbbMiniButtonBaseElement } from '../../core/base-elements.js';
+import { hostAttributes } from '../../core/decorators.js';
+import { setOrRemoveAttribute } from '../../core/dom.js';
+import { SbbDisabledMixin } from '../../core/mixins.js';
+import { AgnosticMutationObserver } from '../../core/observers.js';
+import type { SbbAutocompleteGridOptionElement } from '../autocomplete-grid-option.js';
+
+import style from './autocomplete-grid-button.scss?lit&inline';
+
+let autocompleteButtonNextId = 0;
+
+/** Configuration for the attribute to look at if component is nested in a sbb-optgroup */
+const buttonObserverConfig: MutationObserverInit = {
+  attributeFilter: ['data-group-disabled'],
+};
+
+/**
+ * It displays an icon-only button that can be used in `sbb-autocomplete-grid`.
+ *
+ * @slot icon - Slot used to display the icon, if one is set
+ */
+@customElement('sbb-autocomplete-grid-button')
+@hostAttributes({
+  tabindex: null,
+})
+export class SbbAutocompleteGridButtonElement extends SbbDisabledMixin(SbbMiniButtonBaseElement) {
+  public static override styles: CSSResultGroup = style;
+
+  /** Gets the SbbAutocompleteGridOptionElement on the same row of the button. */
+  public get option(): SbbAutocompleteGridOptionElement | null {
+    return (
+      this.closest('sbb-autocomplete-grid-row')?.querySelector('sbb-autocomplete-grid-option') ||
+      null
+    );
+  }
+
+  /** Whether the component must be set disabled due disabled attribute on sbb-optgroup. */
+  @state() private _disabledFromGroup = false;
```

Comment:
Is @state really needed here? And moreover, could you somehow use the isDisabledExternally logic of the disabledmixin? maybe only the check below could be simplified.
```suggestion
private _disabledFromGroup = false;
```

Reply:
Did some tests, and seems to work with isDisabledExternally